lib/logger.js

```diff
@@ -0,0 +1,393 @@
+    }
+
+    throw new ERR_INVALID_ARG_TYPE('options.stream',
+                                   ['number', 'string', 'Utf8Stream'],
+                                   stream);
+  }
+
+  handle(record) {
+    const logObj = {
+      level: record.level,
+      time: record.time,
+      msg: record.msg,
+      ...this._fields,
+      ...record.bindings,
+      ...record.fields,
+    };
+
+    const json = JSONStringify(logObj) + '\n';
+    this._stream.write(json);
+  }
+
+  /**
+   * Flush pending writes
+   * @param {Function} callback
+   */
+  flush(callback) {
+    this._stream.flush(callback);
+  }
+
+  /**
+   * Flush pending writes synchronously
+   */
+  flushSync() {
+    this._stream.flushSync();
+  }
+
+  /**
+   * Close the consumer
+   */
+  end() {
+    this._stream.end();
+  }
+}
+
+/**
+ * Logger class
+ */
+class Logger {
+  constructor(options = {}) {
+    validateObject(options, 'options');
+    const {
+      level = 'info',
+      bindings = {},
+    } = options;
+
+    validateString(level, 'options.level');
+    if (!LEVELS[level]) {
+      throw new ERR_INVALID_ARG_VALUE('options.level', level,
+                                      `must be one of: ${LEVEL_NAMES.join(', ')}`);
+    }
+
+    validateObject(bindings, 'options.bindings');
+
+    this._level = level;
+    this._levelValue = LEVELS[level];
+    this._bindings = bindings;
+
+    // Optimize: replace disabled log methods with noop
+    this._setLogMethods();
+  }
+
+  /**
+   * Replace disabled log methods with noop for performance
+   * @private
+   */
+  _setLogMethods() {
+    const levelValue = this._levelValue;
+
+    if (levelValue > 10) this.trace = noop;
+    if (levelValue > 20) this.debug = noop;
+    if (levelValue > 30) this.info = noop;
+    if (levelValue > 40) this.warn = noop;
+    if (levelValue > 50) this.error = noop;
+    if (levelValue > 60) this.fatal = noop;
+  }
+
+  /**
+   * Check if a level would be logged
+   * @param {string} level
+   * @returns {boolean}
+   */
+  enabled(level) {
+    return LEVELS[level] >= this._levelValue;
+  }
+
+  /**
+   * Create a child logger with additional context
+   * @param {object} bindings - Context to add to all child logs
+   * @param {object} [options] - Optional overrides
+   * @returns {Logger}
+   */
+  child(bindings, options = {}) {
+    validateObject(bindings, 'bindings');
+    validateObject(options, 'options');
+
+    const mergedBindings = ObjectAssign(
+      { __proto__: null },
+      this._bindings,
+      bindings,
+    );
+
+    return new Logger({
+      level: options.level || this._level,
+      bindings: mergedBindings,
+    });
+  }
+
+  /**
+   * Check if value is an Error object
+   * @param {*} value
+   * @returns {boolean}
+   * @private
+   */
+  _isError(value) {
+    return value !== null &&
+           typeof value === 'object' &&
+           (ObjectPrototypeToString(value) === '[object Error]' ||
+            value instanceof Error);
```

Comment:
Why not `ErrorIsError`?

Reply:
I think this feature is not yet available in Node.js Core

Reply:
> I think this feature is not yet available in Node.js Core

It is part of v8, and exists AFAIK starting from v24?

Reply:
you right!, thanks I finded

Reply:
https://github.com/nodejs/node/pull/60468/commits/1a76b7fa20a23ec51bc80cd0ebf262104dc53cb7

When I tried again it based on @marco-ippolito  suggestion, success. 🚀

Reply:
I was mistaken, https://github.com/nodejs/node/pull/60468#issuecomment-3611485415

I think ErrorIsError not defined in primordials,
I wonder your any suggestions? @marco-ippolito @atlowChemi

Reply:
I used isNativeError thanks @aduh95 (We talked on Slack)